### PR TITLE
Deprecate `auto_select_gpus` and enable it by default

### LINKED ISSUE #13079: RFC: Deprecate  `auto_select_gpus` Trainer argument
## Proposed refactor

Deprecate the `Trainer(auto_select_gpus=True|False)` option and just enable it always when `devices=int` gets selected. 

### Motivation

- The feature is misleading, we don't actually have an algorithm to select the gpus automatically in a smart way. A confusion by user was raised recently: #13012
- The flag only applies to the GPU accelerator. There are no equivalent flags for tpus, ipus, etc.
- What auto_select_gpus does is such a niche use case that it's almost not worth talking about. The implementation just runs through all available GPUs and tests if it can place a tensor in memory. This essentially tests wether the GPU is in exclusive mode or not. [See the implementation here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/tuner/auto_gpu_select.py).
- Internally, the flag is framed as "tuning" as the functions are under the tuner module, but this does not fall under the term "tuning" in my opinion.

### Pitch

Keep the feature, but simply enable it by default and remove the flag from the Trainer. It is still useful to have the check against exclusivity, for example on managed clusters, and I can't think of a reason why it would be undesired. This would only apply when passing `devices=int` and not when indices get passed. 

### Alternatives

Keep it and come up with sophisticated methods to determine whether a GPU should be selected or not based on memory profile, utilization etc. This IMO is not feasible because not enough information is available before training.

### Additional context

#13012

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #13079, #10535

Before
```py
# Picks the first two available GPUs
Trainer(devices=2, auto_select_gpus=True)
```

Now:
```py
# Picks the first two available GPUs
Trainer(devices=2)
```

### Does your PR introduce any breaking changes? If yes, please list them.

Yes. The `auto_select_gpus` argument in Trainer will be removed in v1.9. Users of `auto_select_gpus` can just remove the flag and still make use of it as it is now turned on by default.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/Lightning-AI/lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -517,7 +515,7 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
         self._gpus = self._devices_flag if not self._gpus else self._gpus
         self._tpu_cores = self._devices_flag if not self._tpu_cores else self._tpu_cores
 
-        self._set_devices_flag_if_auto_select_gpus_passed()
+        self._set_devices_flag_if_count_passed()
```
Comment (awaelchli) [inline src/pytorch_lightning/trainer/connectors/accelerator_connector.py:518]: We could merge this with `Accelerator.parse_devices` (see line below). I.e., we would implement `GPUAccelerator.parse_devices` by calling the `pick_multiple_gpus` function there.

File: src/pytorch_lightning/utilities/device_parser.py [hunk 2]
```diff
@@ -49,20 +49,6 @@ def determine_root_gpu_device(gpus: List[_DEVICE]) -> Optional[_DEVICE]:
     return root_gpu
 
 
-def _parse_devices(
```
Comment (awaelchli) [inline src/pytorch_lightning/utilities/device_parser.py:52]: removed dead code

**Comment by carmocca** [comment]
Is this blocked or something?

**Comment by awaelchli** [comment]
This just needs more work. Requires to look at tests etc. Need to discuss priority and then add to sprint.
